beacon_node/execution_layer/src/lib.rs

```diff
@@ -1148,62 +1108,47 @@ impl<T: EthSpec> ExecutionLayer<T> {
                     )));
                 }
 
-                if self.inner.always_prefer_builder_payload {
-                    return ProvenancedPayload::try_from(relay.data.message);
-                }
+                let builder_boost_factor =
+                    builder_boost_factor.unwrap_or(DEFAULT_BUILDER_BOOST_FACTOR);
 
                 let relay_value = *relay.data.message.value();
+                let boosted_relay_value =
+                    relay_value.saturating_mul((builder_boost_factor / 100).into());
```

Comment:
Order of operations is important here, as if we divide the `builder_boost_factor` by 100 first, we'll lose information due to integer rounding. e.g. any value <100 will round down to 0, and values between 100-199 will be rounded to 1.

Dividing the relay value first then multiplying is what the spec says to do:

```suggestion
                    (relay_value / 100).saturating_mul(builder_boost_factor);
```

Reply:
fixed. I also made a change to skip the calculation completely if `builder_boost_factor` is `None`, as `relay_value/100` can also result in some information loss.

Reply:
Nice, that's a good improvement